### PR TITLE
Suggestion History Feed Content Screen (for issue #299)

### PR DESCRIPTION
**Pull request by annhit29**
In this pull request, I
1) established the navigation between the buttons to link to the SuggestionHistoryFeedContent screen. Thus when we go to the suggestion feed content screen, there is the history icon button on the right of the search bar.
2) I created the screen which displays all the suggestion histories. A suggestion history is a suggestion that has been transformed to a stop, which is no longer visible on the suggestion screen. If users want to see the details of a suggestion history, they can go to this screen by clicking on the history icon.
3) Adapt the SuggestionFeedContent screen and the DashboardSuggestionWidget to only display suggestions that are not stops if there are.
4) I created SuggestionHistoryItem that is an archived suggestion item, in which we can no longer modify anything that were modifiable, such as the number of likes, the number of comments (since we can no longer comment on that suggestion of the trip).

\
This pull request is for the issue #299 .

### CODE REVIEW

**Review by Exino329 — CHANGES_REQUESTED** [review]
Good work overall ! There is just few things that should be reconsidering to improve the UI : 
- The history logo could be placed at the bottom of the suggestion feed content (on the same row as the "add suggestion button" on the left) because actually, it is placed at the left of the search bar and it is kinda small. Also placing a button just next to the search bar isn't really intuitive for a user in term of experience
-  When data is loading, There is actually a text indicating that data is loading, however it would be better to use a  `CircularProgressIndicator` which is more appropriate and easy to implement. It is already widely used in our app so the UI for loading state is the same everywhere. (For example, you can find a usage in `Overview.kt`.
- You can add padding between "Suggestion history" text and the history content. Otherwise we have the feeling that when we scroll in the suggestion history, the top content is "sticked" with the lazy column content. You can also make the top content containing the suggestion history text filled with primary theme color and put the text in white (as in the agenda view for example).
- Since We implement an historic, it would be better that the most recent deleted suggestion is at the top of the column in the history. Actually it is placed at the bottom

Also There is a bug to fix : when converting a suggestion to stop on click (not using the like system), the suggestion is not added in the suggestion history.

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/AgendaViewModel.kt [hunk 1]
```diff
@@ -43,14 +43,48 @@ open class AgendaViewModel(
 
   open var selectedDate: LocalDate? = LocalDate.now()
 
+  open val _stopsInfo = MutableStateFlow<Map<LocalDate, CalendarUiState.StopStatus>>(emptyMap())
+
+  /** Exposed read-only state flow of stops info. */
+  //  val stopsInfo: StateFlow<Map<LocalDate, CalendarUiState.StopStatus>> =
```
[the remark this answers is not in the document]
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/model/viewmodel/AgendaViewModel.kt:49]: Thank you, it's deleted now :D

File: app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardSuggestionWidget.kt
[the remark this answers is not in the document]
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/dashboard/DashboardSuggestionWidget.kt:1]: Thank you very much, Lorin!

File: app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionHistoryItem.kt [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+import java.time.format.DateTimeFormatter
+
+/**
+ * Composable function that represents a single suggestion item in the suggestion feed.
+ *
+ * @param suggestion The suggestion object to be displayed.
+ * @param modifier The modifier to be applied to the suggestion item.
+ */
+@SuppressLint("UnrememberedMutableState")
+@Composable
```
[the remark this answers is not in the document]
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionHistoryItem.kt:52]: Thanks, removed!

File: app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionHistoryFeedContent.kt
[the remark this answers is not in the document]
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionHistoryFeedContent.kt:1]: Thank you!
I did this: 
> remove the tripId param of this composable and use suggestionViewModel.tripId

So I removed the `tripId` parameter, declared `val tripId = suggestionsViewModel.tripId` then remove everywhere `SuggestionHistoryFeedContent` is called.

File: app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionHistoryFeedContent.kt [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+    // If suggestion list is empty, display a message
+    if (addedSuggestions.isEmpty()) {
+      Box(modifier = Modifier.fillMaxSize()) {
+        Text(
+            modifier =
+                Modifier.width(260.dp)
+                    .height(55.dp)
+                    .align(Alignment.Center)
+                    .testTag("noSuggestionsHistoryToDisplay"),
```
[the remark this answers is not in the document]
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionHistoryFeedContent.kt:105]: Yes, it's indeed more natural! Done and thank you :D

File: app/src/main/java/com/github/se/wanderpals/ui/screens/trip/agenda/DatesOperations.kt [hunk 4]
```diff
@@ -35,6 +35,17 @@ data class CalendarUiState(
     val Init = CalendarUiState(yearMonth = YearMonth.now(), dates = emptyList())
   }
 
+  /**
+   * Enum class representing the status of the stop addition.
+   *
+   * @property ADDED Suggestion was added to the trip, it is a stop of the trip now.
+   * @property NONE Suggestion has not been added, it is still a suggestion of the trip.
+   */
```
[the remark this answers is not in the document]
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/trip/agenda/DatesOperations.kt:43]: Yes, also please see the response above. It is something we will discuss.

**Comment by annhit29** [comment]
Thank you for the precious review!

> The history logo could be placed at the bottom of the suggestion feed content (on the same row as the "add suggestion button" on the left) because actually, it is placed at the left of the search bar and it is kinda small. Also placing a button just next to the search bar isn't really intuitive for a user in term of experience

Yes, both UI and UX are better now, the button is large enough and it is very easy to find and click on it. The searchbar is at where it was.
image


> When data is loading, There is actually a text indicating that data is loading, however it would be better to use a CircularProgressIndicator which is more appropriate and easy to implement. It is already widely used in our app so the UI for loading state is the same everywhere. (For example, you can find a usage in Overview.kt.

Done, thanks!

> You can add padding between "Suggestion history" text and the history content. Otherwise we have the feeling that when we scroll in the suggestion history, the top content is "sticked" with the lazy column content. You can also make the top content containing the suggestion history text filled with primary theme color and put the text in white (as in the agenda view for example).

I changed from the use of Column composable to Scaffold composable, and I have also added spaces before the list of suggestion histories, and spaces between suggestionHistoryItems
image

> Since We implement an historic, it would be better that the most recent deleted suggestion is at the top of the column in the history. Actually it is placed at the bottom

Yes, I used `asReversed()` function to achieve this.

**Review by Exino329 — APPROVED** [review]
Looks good to me ! Also as discussed concerning the history, ordering the suggestion by transformation to stop date would involve a refactor of the suggestion data class which is not convenient. So instead of this, the suggestion are display in the descending order of creation.

**Review by Exino329 — APPROVED** [review]
LGTM !

**Pull request merged**